dockerfiles/executor_image/Dockerfile

```diff
@@ -4,50 +4,36 @@
 # debian 12
 FROM debian@sha256:fac2c0fd33e88dfd3bc88a872cfb78dcb167e74af6162d31724df69e482f886c
 
+# Install skopeo and umoci which we use to unpack OCI images when we're not using docker.
+# Install iproute2 ("ip" command) to configure networking on host.
 RUN apt-get update && \
     apt-get install -y \
-    coreutils curl apt-transport-https ca-certificates gnupg-agent software-properties-common fuse && \
+		# Install skopeo and umoci which we use to unpack OCI images when we're not using docker.
+		# Install iproute2 ("ip" command) to configure networking on host.
+    fuse skopeo umoci iproute2 amazon-ecr-credential-helper && \
+    apt-get upgrade -y && \
     rm -rf /var/lib/apt/lists/* && apt-get clean
 
-RUN ARCH=$(dpkg --print-architecture) && \
-    SHASUM=$(case $ARCH in \
-        amd64) echo '93dcc18adc78c65a028a84799ecf8ad40c936fdfc5f2a57b1acda5a8117fa82c' ;; \
-        arm64) echo '07952557df20bfd2a95f9bef198b445e006171969499a1d361bd9e6f8e5e0e81' ;; \
-    esac) && \
-    curl -o /tini -fsSL "https://github.com/krallin/tini/releases/download/v0.19.0/tini-${ARCH}" && \
-    echo "$SHASUM /tini" | sha256sum -c && \
-    chmod +x /tini
-
-RUN install -m 0755 -d /etc/apt/keyrings && \
+RUN DOCKER_VERSION="5:26.1.4-1~debian.12~bookworm" && \
+    CONTAINERD_DEB_VERSION="1.7.19-1" && \
+    DOCKER_BUILDX_VERSION="0.15.1-1~debian.12~bookworm" && \
+    apt-get update && \
+    apt-get install -y \
+    curl ca-certificates apt-transport-https && \
+    install -m 0755 -d /etc/apt/keyrings && \
     curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc && \
     chmod a+r /etc/apt/keyrings/docker.asc && \
     echo \
       "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
       $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
     tee /etc/apt/sources.list.d/docker.list > /dev/null && \
-    apt-get update && apt-get install -y docker-ce docker-ce-cli containerd.io && \
-    rm -rf /var/lib/apt/lists/* && apt-get clean
-
-# Install skopeo and umoci which we use to unpack OCI images when we're not using docker.
-# Install iproute2 ("ip" command) to configure networking on host.
-RUN apt-get update && \
-    apt-get install -y skopeo umoci iproute2 amazon-ecr-credential-helper && \
+    apt-get update && apt-get install -y \
+    docker-ce=${DOCKER_VERSION} \
+    docker-ce-cli=${DOCKER_VERSION} \
+    containerd.io=${CONTAINERD_DEB_VERSION} \
```

Comment:
It looks like we're overwriting certain binaries via image layers - I think that will increase the image size significantly? might make sense to either overwrite those files in this step (instead of adding an overwrite layer) or just delete the files in this step (and sprinkle some comments that these deletions should be kept in sync with the container_layer)